src/components/extracrewdetails.tsx

```diff
@@ -37,9 +64,94 @@ class ExtraCrewDetails extends Component<ExtraCrewDetailsProps, ExtraCrewDetails
 		}
 
 		let self = this;
+		fetch('/structured/keystones.json')
+			.then(response => response.json())
+			.then(allkeystones => {
+				let crew_keystone_crate = allkeystones.find((k: any) => k.crew_archetype_id === this.props.crew_archetype_id);
+
+				if (crew_keystone_crate) {
+					self.setState({
+						keystone: {
+							name: crew_keystone_crate.name, flavor: crew_keystone_crate.flavor,
+							keystones: crew_keystone_crate.keystones.map((kid: any) => allkeystones.find((k: any) => k.id === kid))
+						}
+					});
+				}
+			});
+
 		fetch('/structured/crew.json')
 			.then(response => response.json())
 			.then(crew => {
+				// Find all crew who have any polestars in common
+				for (let i = 0; i < crew.length; i++) {
+					let polesInCommon = [];
+					for (let t = 0; t < this.props.traits.length; t++) {
+						if (crew[i].traits.indexOf(this.props.traits[t]) >= 0)
+							polesInCommon.push(this.props.traits[t]);
+					}
+					// Add 1 to count of every polestar combination in common
+					if (polesInCommon.length > 0) {
+						if (crew[i].max_rarity == this.props.max_rarity)
+							polesInCommon.push('crew_max_rarity_'+this.props.max_rarity);
+						for (let skill in crew[i].base_skills) {
+							if (crew[i].base_skills[skill] && this.props.base_skills[skill])
+								polesInCommon.push(skill);
+						}
+						polestarCounts.forEach(count => {
+							if (polesInCommon.length >= count.polestars.length) {
+								let bMatching = true;
+								count.polestars.forEach(polestar => {
+									bMatching = bMatching && polesInCommon.indexOf(polestar) >= 0;
+								});
+								if (bMatching) count.count++;
+							}
+						});
+					}
+				}
+				// Find optimal polestars, i.e. combinations with best chance of retrieving this crew
+				polestarCounts.sort((a, b) => {
+					if (a.count == b.count)
+						return a.polestars.length > b.polestars.length;
+					return a.count > b.count;
+				});
+				let optimized = [], iBestCount = 5, iMinimumTraits = 4;
```

Comment:
We probably need a higher minimum, at least 10. For example, Alternate Future T'Pol best combo is shared with 8 crew. 20 possibly even better.

```suggestion
				let optimized = [], iBestCount = 10, iMinimumTraits = 4;
```